lib/header.c

```diff
@@ -474,8 +474,8 @@ static int dataLength(rpm_tagtype_t type, rpm_constdata_t p, rpm_count_t count,
     default:
 	if (typeSizes[type] == -1)
 	    return -1;
-	length = typeSizes[(type & 0xf)] * count;
-	if (length < 0 || (se && (s + length) > se))
+	length = typeSizes[type] * count;
```

Comment:
The `& 0xf` was probably there to guarantee that we're accessing typeSizes within valid range.
Shouldn't we rather check the `type` range with the following? `if (type > RPM_MAX_TYPE) return -1;`